pkg/reconciler/namespace/controller.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package namespace
+
+import (
+	"context"
+
+	nsinformer "knative.dev/pkg/injection/informers/kubeinformers/corev1/namespace"
+	kcertinformer "knative.dev/serving/pkg/client/injection/informers/networking/v1alpha1/certificate"
+
+	"k8s.io/client-go/tools/cache"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/serving/pkg/apis/networking"
+	"knative.dev/serving/pkg/network"
+	"knative.dev/serving/pkg/reconciler"
+	"knative.dev/serving/pkg/reconciler/namespace/config"
+)
+
+const (
+	controllerAgentName = "namespace-controller"
+)
+
+type configStore interface {
+	ToContext(ctx context.Context) context.Context
+	WatchConfigs(w configmap.Watcher)
+}
+
+// NewController initializes the controller and is called by the generated code
+// Registers eventhandlers to enqueue events.
+func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	nsInformer := nsinformer.Get(ctx)
+	knCertificateInformer := kcertinformer.Get(ctx)
+
+	c := &Reconciler{
+		Base:                reconciler.NewBase(ctx, controllerAgentName, cmw),
+		nsLister:            nsInformer.Lister(),
+		knCertificateLister: knCertificateInformer.Lister(),
+	}
+
+	impl := controller.NewImpl(c, c.Logger, "Namespace")
+
+	c.Logger.Info("Setting up event handlers")
+	nsInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: reconciler.LabelExistsFilterFunc(networking.WildcardCertLabelKey),
+		Handler:    controller.HandleAll(impl.Enqueue),
+	})
+
+	knCertificateInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: reconciler.LabelExistsFilterFunc(networking.OwnerNamespaceLabelKey),
+		Handler:    controller.HandleAll(impl.EnqueueLabelOfClusterScopedResource(networking.OwnerNamespaceLabelKey)),
```

Comment:
Why isn't this just using `EnqueueControllerOf` and using the `Namespace` in an `OwnerReference`?